libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -515,22 +516,41 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
     };
     
     // Create handler
-    SFSyncServerTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
+    SFSyncUpTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
         // Replace id with server id during create
         record[kId] = d[kSyncManagerLObjectId];
         completeBlockUpdate(d);
     };
     
-    SFSyncServerTargetCompleteBlock completeBlock;
+    
+    SFSyncUpTarget *target = (SFSyncUpTarget*) sync.target;
+    NSArray *fieldList = sync.options.fieldlist;
+    
+    // Getting type and id
+    NSString* objectType = [SFJsonUtils projectIntoJson:record path:kObjectTypeField];
+    NSString* objectId = record[kId];
+    
+    // Fields to save (in the case of create or update)
+    NSMutableDictionary* fields = [NSMutableDictionary dictionary];
```

Comment:
*&gt; won't we simply end up with the same code in all custom targets (looping through the fieldlist, and projecting out the chosen fields)?*

Well sure, but that's the idea: it ends up in the *target*, which (in my thinking anyway) should have ownership over what defines the structure of the data it's sending.

Not sure about the "" part—isn't `[record objectForKey@""]` going to return `nil`?

Anyway, I don't know that the switch in this model will be too restrictive for our use cases.  I do think that merging targets and options could make sense, which would be another path into that flexibility.  But this is fine at this point...we can see how it works in our consumption model, and ping you guys with our results.